prov/rxm/src/rxm_init.c

```diff
@@ -91,11 +91,11 @@ int rxm_info_to_core(uint32_t version, const struct fi_info *hints,
 	core_info->mode |= FI_RX_CQ_DATA | FI_CONTEXT;
 
 	if (hints) {
-		if (hints->caps & FI_TAGGED)
+		if (hints->caps & (FI_ATOMIC | FI_TAGGED))
 			core_info->caps |= FI_MSG;
 
 		/* FI_RMA cap is needed for large message transfer protocol */
-		if (hints->caps & (FI_MSG | FI_TAGGED))
+		if (hints->caps & (FI_MSG | FI_ATOMIC | FI_TAGGED))
```

Comment:
FI_ATOMIC can be skipped here as the dependency tree is FI_ATOMIC -> FI_MSG -> FI_RMA

Reply:
If I had requested only FI_ATOMIC capability in the hints, are not both needed? I think the first test would test bit FI_ATOMIC which will add FI_MSG to the core info, then the second test would test FI_ATOMIC which will add FI_RMA (i.e. the result of the first test doesn't update the hints being tested).

Reply:
Should the first assignment be caps |= hints->caps & PASSTHRU, or simply '='.  Using |= implies that the caps have been set earlier in the code path, whereas = makes it clear that this is the first time caps is initialized.

Reply:
Although no one sets `core_info->caps` before, I think having `|=` would allow for such possibility in the future. What do you think?

Reply:
> I see. It looks like the current code is not exactly right - oops.
> 
> The checks need to be modified as:
> 
> ```
> #define RXM_PASSTHRU_CAPS (FI_MSG | FI_RMA | FI_SEND | FI_RECV | FI_READ | FI_WRITE | FI_REMOTE_READ | FI_REMOTE_WRITE)
> 
> core_info->caps |= hints->caps & RXM_PASSTHRU_CAPS);
> if (hints->caps & (FI_ATOMIC | FI_TAGGED)
>     core_info->caps |= FI_MSG;
> if (core_info->caps & FI_MSG)
>     core_info->caps |= FI_RMA;
> ```

Ok, that makes sense knowing you meant to test core_info when setting RMA in the current code. I will update the PR to reflect these changes. Thanks!

Reply:
Well, having the |= confused me because I thought caps was being initialized earlier in the code path, but I couldn't find where.  I would drop it and be clear that caps is being initialized here.

Reply:
👍

Reply:
Done, and to just "=" on the initial setting. Thanks.